packages/Slider/Range.tsx

```diff
@@ -0,0 +1,276 @@
+  ({
+    borderSelectorColor = 'light.900',
+    disabled,
+    hint,
+    label,
+    max,
+    min,
+    onChange,
+    step = 1,
+    tooltip,
+    type,
+    value,
+    values,
+    ...restProps
+  }) => {
+    const [minValue, setMinValue] = useState(min)
+    const [maxValue, setMaxValue] = useState(max)
+    const minValueRef = useRef<HTMLInputElement>(null)
+    const maxValueRef = useRef<HTMLInputElement>(null)
+    const range = useRef<HTMLDivElement>(null)
+    const tooltipMinRef = useRef<HTMLOutputElement>(null)
+    const tooltipMaxRef = useRef<HTMLOutputElement>(null)
+    const [tooltipMinVisible, setTooltipMinVisible] = useState<boolean>(false)
+    const [tooltipMaxVisible, setTooltipMaxVisible] = useState<boolean>(false)
+
+    const handleMinValue = (e: ChangeEvent<HTMLInputElement>) => {
+      // Prevents the min value from being above the max
+      const value = round(Math.min(parseInt(e.target.value, 10), maxValue - 1), step)
+      setMinValue(value)
+      e.target.value = value.toString()
+    }
+
+    const handleMaxValue = (e: ChangeEvent<HTMLInputElement>) => {
+      // Prevents the max from being below the min
+      const value = round(Math.max(parseInt(e.target.value, 10), minValue + 1), step)
+      setMaxValue(value)
+      e.target.value = value.toString()
+    }
+
+    const getPercent = useCallback(
+      (value: number) => {
+        const percent = Math.round(((value - min) / (max - min)) * 100)
+        return percent > max ? max : percent < min ? min : percent
+      },
+      [min, max]
+    )
+
+    // When minValue changes we have to decrease the left side range
+    useEffect(() => {
+      // Using ref instead of state value because we want to avoid another new dependency in the useEffect (we only want to change range width when minValue changes)
+      if (maxValueRef.current) {
+        const minPercent = getPercent(minValue)
+        const maxPercent = getPercent(parseInt(maxValueRef.current.value, 10))
+
+        if (range.current) {
+          range.current.style.left = `${minPercent}%`
+          range.current.style.width = `${maxPercent - minPercent}%`
+        }
+      }
+    }, [minValue, getPercent])
+
+    // When maxValue changes we have to decrease the right side range
+    useEffect(() => {
+      // Same as below but for minValue
+      if (minValueRef.current) {
+        const minPercent = getPercent(parseInt(minValueRef.current.value, 10))
+        const maxPercent = getPercent(maxValue)
+
+        if (range.current) {
+          range.current.style.width = `${maxPercent - minPercent}%`
+        }
+
+        if (tooltipMinRef.current) {
+          const thumbWidth = 20
+          const fraction = getPercent(minValue) / 100
+          tooltipMinRef.current.style.left = `calc(${fraction * 100}% + ${
+            (0.5 - fraction) * thumbWidth
+          }px)`
+        }
+
+        if (tooltipMaxRef.current) {
+          const thumbWidth = 20
```

Comment:
Maybe you could extract this const above the component? I don't know if there is another const like this one, which does not rely on any variable, but you can extract them from the component render.